torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3174,11 +3174,32 @@ struct to_ir {
           return dim + 1;
         }
       }
-      TypePtr type_hint = OptionalType::ofTensor();
-      if (subscript_expr.kind() == TK_NONE) {
-        type_hint = NoneType::get();
+      Value* index;
+      if (subscript_expr.kind() == TK_LIST_LITERAL) {
```

Comment:
One question, maybe as a followup to this PR: seems like we should be able to handle any type of list, not just a list literal? e.g., the follow should work:
```
def f(x):
    ls = [0]
    ls.append(1)
    ls.append(2)
    return x[ls]
```

Reply:
Yea good point, it should be possible (maybe by unwrapping a list out of the TK_VAR). I'll take a look and send a followup PR if it works. :D